core/field_dropdown.ts

```diff
@@ -114,30 +111,29 @@ export class FieldDropdown extends Field {
    * @throws {TypeError} If `menuGenerator` options are incorrectly structured.
    */
   constructor(
-      menuGenerator: AnyDuringMigration[][]|Function|Sentinel,
-      opt_validator?: Function, opt_config?: FieldConfig) {
+      menuGenerator: MenuGenerator, opt_validator?: Function,
+      opt_config?: FieldConfig);
+  constructor(menuGenerator: Sentinel);
+  constructor(
+      menuGenerator: MenuGenerator|Sentinel, opt_validator?: Function,
+      opt_config?: FieldConfig) {
     super(Field.SKIP_SETUP);
 
-    // If we pass SKIP_SETUP, don't do *anything* with the menu generator.
-    if (menuGenerator === Field.SKIP_SETUP) {
+    if (isMenuGenerator(menuGenerator)) {
+      if (Array.isArray(menuGenerator)) {
+        validateOptions(menuGenerator);
+        const trimmed = trimOptions(menuGenerator);
```

Comment:
Before, `trimOptions` was mutating the deep copied menuGenerator and, when no `ImageProperties` labels were provided, setting a new value for `menuGenerator_`. Assuming there's not an edge case I'm missing where the mutation is needed, this update shifts setting the value for `menuGenerator_` to one point, and it should eliminate the need for a deep copy.